src/coreclr/src/jit/lsra.cpp

```diff
@@ -7104,9 +7103,9 @@ void LinearScan::resolveRegisters()
                     Interval* localVarInterval = interval->relatedInterval;
                     if ((localVarInterval->physReg != REG_NA) && !localVarInterval->isPartiallySpilled)
                     {
-                        // If the localVar is in a register, it must be a callee-save register (otherwise it would have
-                        // already been spilled).
-                        assert(localVarInterval->assignedReg->isCalleeSave);
+                        // If the localVar is in a register, it must be in a register that is not trashed by
+                        // the current node (otherwise it would have already been spilled).
+                        assert((genRegMask(localVarInterval->physReg) & getKillSetForNode(treeNode)) == RBM_NONE);
```

Comment:
`GT_PROF_HOOK` has a special calling convention that doesn't kill the argument registers, so they may still be live here, so instead of checking that its in a callee-save, we need to check that it's *not* in the kill set for the current node.